src/hooks/web3Context.tsx

```diff
@@ -165,10 +170,10 @@ export const Web3ContextProvider: React.FC<{ children: ReactElement }> = ({ chil
     // Eventually we'll be fine without doing network validations.
     setAddress(connectedAddress);
     setProvider(connectedProvider);
-    _initListeners();
```

Comment:
You still want to run this for walletProviders. Why remove it?

Reply:
Oh i see the useEffect down below. Makes sense. It also means it will be run when connected goes from true --> false, but you added a defensive check above so that makes sense as well. good shit.

Reply:
Yea I had to move `_initListeners()` out of `connect` because state isn't updated synchronously. It's not updated until the useEffect hook down below.